mqtt/mqtt-broker/src/broker.rs

```diff
@@ -100,6 +104,44 @@ where
         Ok(self.snapshot())
     }
 
+    fn reevaluate_subscriptions(&mut self) -> Result<(), Error> {
+        let mut client_ids_to_remove: Vec<ClientId> = Vec::new();
+        for session in &self.sessions {
+            for sub in session.1.subscriptions().clone().unwrap().values() {
+                let client_info = session.1.client_info()?.clone();
+                let operation = Operation::new_subscribe(proto::SubscribeTo {
+                    topic_filter: sub.filter().clone().to_string(),
+                    qos: *sub.max_qos(),
+                });
+                let activity = Activity::new(session.0.clone(), client_info, operation);
+                match self.authorizer.authorize(activity) {
+                    Ok(Authorization::Allowed) => {
+                        debug!("client {} successfully reauthorized", session.0);
```

Comment:
nit: I would use the same log messages we use in normal flow. I would just add another log entry at the beginning of the method.

Reply:
There is different behavior here, though.
If a client originally tries to subscribe and is forbidden, they can move on and try other subscriptions.
But if a client reauthorization fails for a subscription, the whole client must be disconnected. Since it's different behavior, I think different logs would be helpful